addon/index.js

```diff
@@ -77,6 +75,51 @@ function updatePrefs() {
   });
 }
 
+// Listen for metrics events from experiments.
+const metrics = {
+  isInitialized: false,
+  init: function() {
+    if (this.isInitialized) {
+      return;
+    }
+    this.isInitialized = true;
+    // Note: the observer service holds a strong reference to this observer,
+    // so we must detach it on shutdown / uninstall by calling destroy().
+    Services.obs.addObserver(metrics, 'idea-town::send-metric', false);
+  },
+  destroy: function() {
+    Services.obs.removeObserver(metrics, 'idea-town::send-metric', false);
+    this.isInitialized = false;
+  },
+  // The metrics object format is { key, value, addonName }, where
+  //   'key' is the name of the event,
+  //   'value' is the value of the event (can be any JSON-serializable object),
+  //   'addonName' is the name of the experiment sending the data.
+  observe: function() {
+    // The nsIObserverService sends non-useful positional arguments; the third
+    // is the only one we need.
+    const data = arguments[2];
+
+    let d;
+    try {
+      d = JSON.parse(data);
+    } catch (ex) {
+      console.error('Idea Town metrics error: cannot process event, ' +
+                    ' JSON.parse failed: ', ex);
```

Comment:
> That doesn't really answer how we could have 2-way communication between addons, though, beyond fire-and-forget events.

In a hilarious plot twist, I've figured out that it _is_ actually possible to have a single JS file that's compatible with the SDK or the Gecko-centric JSM code loaders. Getting the URI to this file for Cu.import() loading is unforgivably strange, but we can give implementers a snippet to copy-and-paste as they see fit.

Cu.import requires getting the module path relative to the top level of the addon, so a separate addon-api folder still can't work, but we can avoid nsIObserver and just call a callback if an error occurs.

Working on the updated PR now.